dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -371,4 +407,106 @@ public boolean isEmpty() {
         }
     }
 
+    static class PreparedJobContext{
+
+        private final String indexName;
+        private final String newIndexName;
+        private final boolean indexAll;
+        private final String joinedHosts;
+        private final boolean incremental;
+        private final Date startDate;
+        private final Date endDate;
+        private final String jobId;
+        private final String jobName;
+        private final String langList;
+        private final String paths;
+        private final boolean pathInclude;
+        private final List<SiteSearchConfig> configs;
+
+        PreparedJobContext(
```

Comment:
perfect candidate for a builder

Reply:
yeah, I thought so too, but I was refactoring to make this thread-safe.. So at the end, I would end up replacing this call 
`return new PreparedJobContext(indexName, newIndexName, indexAll, joinedHosts, incremental,
                 startDate, endDate, jobId, jobName, langList, pathsAsString, include,
                 builder.build()
         );` to the constructor by a builder ?? 

So all I needed was an immutable method to return data. But if you think it is I'll put it in a builder